### PR TITLE
enhancement(tests): e2e logs ingestion from Datadog agent to Vector

### PR DESCRIPTION
**Pull request by prognant**
Quick summary:
* Add the ability to spawn a generic chart into the k8s e2e tests framework
* Spawn a datadog agent, a vector aggregator and a dummy pod, the datadog agent read container log, ship those to vector where delivery is asserted in a simple test

Side note:
./lib/k8s-test-framework/src/vector.rs may be rename `chart.rs` as it now manage charts on a general basis on not only vector.

### CODE REVIEW

File: lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs
Comment (spencergilbert) [inline lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs:20]:
```suggestion
        address: "0.0.0.0:8080"
        rawConfig: |
          address = "0.0.0.0:8080"
```

Since rawConfig is "deprecated" we may just want to use the newer style here

**Review by spencergilbert — APPROVED** [review]
Just one nitpick

File: lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs
Comment (JeanMertz) [inline lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs:171]: Any reason to drop these manually? They should just drop implicitly at the end of the scope, right?

File: lib/k8s-test-framework/src/vector.rs
Comment (JeanMertz) [inline lib/k8s-test-framework/src/vector.rs:39]: nit: `for` loops implicitly call `into_iter`, so line 38 shouldn't be needed.

**Review by JeanMertz — APPROVED** [review]
LGTM!

**Comment by prognant** [comment]
All comments are accounted for ! Thanks for the reviews, waiting for the CI before merging.

File: lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs
Comment (prognant) [reply lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs]: From what I just tested & checked I don't think there is any valid reason to keep those lines. FYI I borrowed the initial code from https://github.com/timberio/vector/blob/master/lib/k8s-e2e-tests/tests/vector.rs#L111-L113 thus there might be a reason I'm missing that *possible* explains those manual `drop`.

Comment (JeanMertz) [reply lib/k8s-e2e-tests/tests/vector-dd-agent-aggregator.rs]: I _guess_ it's to be more explicit. I don't mind keeping it in, I just wasn't sure if it was done to be explicit, or just a misunderstanding of how `drop` works.

**Pull request merged**
